<launch>
  <!-- args = list of led id -->
	<node name="usb_cam" pkg="usb_cam" type="usb_cam_node" required="true" output="screen">
		<param name="video_device" value="/dev/video0" />
		<param name="image_width" value="640" />
		<param name="image_height" value="480" />
		<param name="pixel_format" value="yuyv" />
		<param name="camera_frame_id" value="usb_cam" />
		<param name="io_method" value="mmap" />
		<param name="autoexposure" value="false" />
        <param name="exposure" value="1"/>
        />
	</node>
<!-- 	<node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/usb_cam/image_raw"/>
    <param name="autosize" value="true" />
  </node> -->

	<node name="image_proc" pkg="image_proc" type="image_proc" ns="usb_cam" required="true">
	
	</node>

	<!-- <node name="colorgui" pkg="cmvision" type="colorgui" respawn="true" output="screen"> 
		<remap from="image" to="/usb_cam/image_raw" />
 	</node>  -->
	<node name="cmvision" pkg="cmvision" type="cmvision" output="screen" required="true">
		<remap from="image" to="usb_cam/image_rect_color" />
		<param name="color_file" value="$(find detect_roof_rectangles)/colors.txt" />
		<param name="mean_shift_on" value="true" />
		<param name="color_radius_pix" value="0" />
		<param name="spatial_radius_pix" value="0" />
		<param name="debug_on" value="false" />
	</node>
	<node pkg="detect_roof_rectangles" type="detect" name="detect"/>

  <node pkg="fake_gps" type="fake_gps_2d_node" name="fake_gps_2d" args="0 1 2">
    <param name="map_file" value="$(find fake_gps)/config/FEATURE_FILES/map.txt" />
    <param name="camera_file" value="$(find fake_gps)/config/FEATURE_FILES/camera.txt" />
    <param name="start_frame" value="/world" />
    <param name="end_frame" value="/camera"/>
  </node>
    <node pkg="tf" type="static_transform_publisher" name="ftsensor_FRAME" args="0 0 0 0 0 0 /world /Features_markers 500" />
<param name="motor_serial_port" value="/dev/ttyUSB2" />
        <param name="motor_baud_rate" value="9600" />
        <param name="servo_serial_port" value="/dev/ttyUSB3" />
        <param name="servo_baud_rate" value="115200" />
        <node pkg="manual_control" type="manual_control_node" name="manual_control" output="screen"/>
</launch>
